crates/holochain/src/conductor/api/api_external/admin_interface.rs

```diff
@@ -102,12 +102,9 @@ impl AdminInterfaceApi for RealAdminInterfaceApi {
                 }
                 let hash = dna.dna_hash().clone();
                 let dna_list = self.conductor_handle.list_dnas().await?;
-                if dna_list.contains(&hash) {
-                    return Err(ConductorApiError::DnaReadError(
-                        "Given dna has already been registered".to_string(),
-                    ));
+                if !dna_list.contains(&hash) {
+                    self.conductor_handle.install_dna(dna).await?;
                 }
-                self.conductor_handle.install_dna(dna).await?;
                 Ok(AdminResponse::DnaRegistered(hash))
```

Comment:
Perhaps in the future we'll want to return a bool indicating whether it was already registered or not, like `HashMap::insert`. But that would be a breaking change to JS conductor-api, so no need now.